cocos/tween/actions/action.ts

```diff
@@ -45,6 +45,7 @@ export class Action {
 
     protected originalTarget: Node | null = null;
     protected target: Node | null = null;
+    public workerTarget: Node | null = null;
```

Comment:
And what's the difference between `workerTarget ` and `target`?

Reply:
`workerTarget` is the actual target tween action runs on when the target initialized in `sequence`, `parallel` or `then` is differernt from the root target. For example:

```ts
tween(this.node).parallel( // this.node is the root target, start method will assign this.node as `target`
  tween(this.node.getComponent(UITransform))).to(XXXXX), // UITransform target is different from `this.node`, so it's the actual target this tween wants to run on. I call it `workerTarget`
  tween(this.node.getComponent(UIOpacity))).to(XXXXX)
)
.start()
```

When `start` is inovked, only the root target `this.node` will be set to every internal actions in the old code, so any other targets in `sequence`, `parallel` and `then` will be ignored. 

I added a new property `workerTarget` to support this feature. But yes, there is probably a better name for it. Maybe `proxyTarget` or just `subTarget`.

Reply:
Or to make it clearer, `targetToApply`? 
If developer apply the same root target for different sub tweens, then `targetToApply` is the root target.
```ts
const targetToApply =  this.targetToApply ?? this.target; // targetToApply may be null if all sub tweens have the same targets.
```

Reply:
What will happen if workerTarget is not a component, but a node, for example
```typescript
tween(this.node).parallel( // this.node is the root target, start method will assign this.node as `target`
  tween(this.node).to(XXXXX),
  tween(this.node.getComponent(UIOpacity))).to(XXXXX)
)
.start()
```

Reply:
In this example, 
```ts
tween(this.node).parallel(   // this.node is the target for root tween, target = this.node, workerTarget = null
  tween(this.node).to(XXXXX), // target = this.node, workerTarget = this.node
  tween(this.node.getComponent(UIOpacity))).to(XXXXX)  // target = this.node, workerTarget = this.node
)
.start()
```
In fact, each tween owns a target, the targets in sub-tween of `parallel/sequence` will be lost since the parallel/sequence only pick up the internal actions of tween stores.

Reply:
And since the target type is a generic `T`, it could be any object types, not only `Node`/`Component`.

Reply:
But the declaration type is `Node | null`.

Reply:
I think it's `unknown` type now.  T for Tween<T> will contain generic type, while passing to Action, it's an unknown type.

Construct Tween<T> with T or Tween.clone/target with <U>.
```ts
class Tween<T extends object = any> {
    constructor(target?: T | null) {...}
    clone<U extends object = any> (target: U): Tween<U> { ... }
    target<U extends object = any> (target: U): Tween<U> { ... }
}
```
When we start a tween, target with T will be passed to ActionManager:

```ts
addAction<T> (action: Action | null, target: T, paused: boolean): void {
    ...
    action.startWithTarget(target); // It will call action's startWithTarget<T>(target)
}

startWithTarget<T> (target: T | null): void {
        this.originalTarget = target;
        this.target = target;  // We use unknown type to store target
    }

    protected originalTarget: unknown = null;
    protected target: unknown = null;
    public workerTarget: unknown = null;

```

In tween-actions.ts, since we need to recover the `T` for the callback ( onStart/onUpdate/onComplete) , we convert unknown to T.
```ts
    startWithTarget<U> (target: U | null): void {
        const isEqual: TypeEquality<T, U> = true;
        if (!isEqual) return;
        super.startWithTarget(target);

        const workerTarget = (this.workerTarget ?? this.target) as T;
        if (!workerTarget) return;
        ......
        if (this._opts!.onStart) { this._opts!.onStart(workerTarget); }
```

```ts
update (t: number): void {
        const workerTarget = this.workerTarget ?? this.target;
        if (!workerTarget) return;
        ......
        if (opts.onUpdate) { opts.onUpdate(workerTarget as T, t); }
        if (t === 1 && opts.onComplete) { opts.onComplete(workerTarget as T); }
```